pkg/netpol/ingressanalyzer/ingress_analyzer.go

```diff
@@ -146,6 +153,49 @@ func getRouteServices(rt *ocroutev1.Route) ([]string, error) {
 	return targetServices, nil
 }
 
+// ///////////////////////////////////////////////////////////////////////////////////////////////
+// k8s Ingress objects analysis
+
+const (
+	defaultBackendError = "default backend kind error"
+	ruleBackendError    = "rule's backend kind error"
+)
+
+func (ia *IngressAnalyzer) mapk8sIngressToServices(ing *netv1.Ingress) error {
+	services, err := getk8sIngressServices(ing)
+	if err != nil {
+		return err
+	}
+	if _, ok := ia.k8sIngressToServicesMap[ing.Namespace]; !ok {
+		ia.k8sIngressToServicesMap[ing.Namespace] = make(map[string][]string)
+	}
+	ia.k8sIngressToServicesMap[ing.Namespace][ing.Name] = services
+	return nil
+}
+
+func getk8sIngressServices(ing *netv1.Ingress) ([]string, error) {
+	ingressStr := types.NamespacedName{Namespace: ing.Namespace, Name: ing.Name}.String()
+	backendServices := make([]string, 0)
+	// if DefaultBackend is provided add its service name to the result
+	if ing.Spec.DefaultBackend != nil {
+		if ing.Spec.DefaultBackend.Service == nil { // i.e backend.Resource is not nil
+			// Currently, only IngressBackend with Service is supported
+			return nil, errors.New(scan.Ingress + " " + ingressStr + ": " + defaultBackendError)
+		}
+		backendServices[0] = ing.Spec.DefaultBackend.Service.Name
+	}
+	// add service names from the Ingress rules
+	for _, rule := range ing.Spec.Rules {
+		for _, path := range rule.IngressRuleValue.HTTP.Paths {
+			if path.Backend.Service == nil {
+				return nil, errors.New(scan.Ingress + " " + ingressStr + ": " + ruleBackendError)
```

Comment:
- let's avoid errors, just ignore rules that are not supported or not containing expected values.
- what about specified ports on a certain rule?

Reply:
> let's avoid errors, just ignore rules that are not supported or not containing expected values.

done

Reply:
> what about specified ports on a certain rule?

I thought at this version we'll assume the ingress rule with a service has the correct port (the same of the service's port)
will update

Reply:
it's also possible that the pod exposes multiple ports and only one of them is specified in the ingress rule.

Reply:
please add logger warnings when ignoring certain rules or resources